qiskit/algorithms/eigensolvers/vqd.py

```diff
@@ -0,0 +1,512 @@
+        result.cost_function_evals = []
+        result.optimizer_time = []
+        result.eigenvalues = []
+
+        if aux_operators is not None:
+            aux_values = []
+
+        for step in range(1, self.k + 1):
+
+            self._eval_count = 0
+            energy_evaluation = self.get_energy_evaluation(
+                step, operator, prev_states=result.optimal_parameters
+            )
+
+            start_time = time()
+
+            # TODO: add gradient support after FidelityGradients are implemented
+            if callable(self.optimizer):
+                opt_result = self.optimizer(  # pylint: disable=not-callable
+                    fun=energy_evaluation, x0=initial_point, bounds=bounds
+                )
+            else:
+                opt_result = self.optimizer.minimize(
+                    fun=energy_evaluation, x0=initial_point, bounds=bounds
+                )
+
+            eval_time = time() - start_time
+
+            result.optimal_point.append(opt_result.x)
+            result.optimal_parameters.append(dict(zip(self.ansatz.parameters, opt_result.x)))
+            result.optimal_value.append(opt_result.fun)
+            result.cost_function_evals.append(opt_result.nfev)
+            result.optimizer_time.append(eval_time)
+            result.eigenvalues.append(opt_result.fun + 0j)
+
+            if aux_operators is not None:
+                bound_ansatz = self.ansatz.bind_parameters(result.optimal_point[-1])
+                # TODO: replace this by the new eval_observables once merged
+                aux_value = self._eval_observables(self.estimator, bound_ansatz, aux_operators)
+                aux_values.append(aux_value)
+
+            if step == 1:
+
+                logger.info(
+                    "Ground state optimization complete in %s seconds.\n"
+                    "Found opt_params %s in %s evals",
+                    eval_time,
+                    result.optimal_point,
+                    self._eval_count,
+                )
+            else:
+                logger.info(
+                    (
+                        "%s excited state optimization complete in %s s.\n"
+                        "Found opt_params %s in %s evals"
+                    ),
+                    str(step - 1),
+                    eval_time,
+                    result.optimal_point,
+                    self._eval_count,
+                )
+
+        # To match the signature of NumpyEigenSolver Result
+        result.eigenvalues = np.array(result.eigenvalues)
+        result.optimal_point = np.array(result.optimal_point)
+        result.optimal_value = np.array(result.optimal_value)
+        result.cost_function_evals = np.array(result.cost_function_evals)
+        result.optimizer_time = np.array(result.optimizer_time)
+
+        if aux_operators is not None:
+            result.aux_operator_eigenvalues = aux_values
+
+        return result
+
+    def get_energy_evaluation(
```

Comment:
We don't need this to be public method right (the equivalent in VQE is no more) As in keep the public API to what we need/want to expose support.